src/main.rs

```diff
@@ -219,21 +217,37 @@ impl Application for Halloy {
                     return Command::none();
                 };
 
-                let command = dashboard.update(
+                let (command, event) = dashboard.update(
                     message,
                     &mut self.clients,
                     &mut self.servers,
                     &mut self.theme,
                     &self.version,
                     &self.config,
                 );
+
                 // Retrack after dashboard state changes
                 let track = dashboard.track();
 
-                Command::batch(vec![
+                if let Some(event) = event {
+                    match event {
+                        dashboard::Event::ReloadConfiguration => {
+                            // TODO: 
+                            // 1. show error screen if fails.
```

Comment:
1. Would be nice if we could flash a quick message and have it dissapear? Not sure we need to take over the whole screen vs just briefly notifying the user.
2. `self.servers` is what drives the servers. So any _new_ server in the config that isn't in `self.servers` can be added and that server will auto-magically connect / show up (we return `self.servers` in `Application::subscription` to create the stream subscription per server). Existing servers, I don't think we _should_ do anything w/ them as that'd cause us to reconnect, is that desired?